src/core/inspect/kernel.rs

```diff
@@ -77,6 +77,12 @@ impl KernelInspector {
             symbols.insert(u64::from_str_radix(data[0], 16)?, String::from(symbol));
         }
 
+        // If all symbols have a 0-address, only the last one will be left in
+        // the map after the above.
+        if symbols.len() == 1 {
+            bail!("Retis likely does not have the rights to read the symbol addresses from /proc/kallsyms.");
```

Comment:
isn't this covered here?

https://github.com/retis-org/retis/pull/209/commits/2084d544c26f187f70e9913d39335ce2920b8960#diff-2bd2e0b9ebfea0d943f831135f8b9eeeaed40507afcfa7303ac2fe64382b4a02R29

Reply:
I see.  The only problem with this is that could be a bit misleading. One might wonder what use case this is needed for.
I personally failed to quickly find an uncovered case for the previous check, but I didn't really try hard :)
Let's leave it as it is, then, it's not harmful :)